fhir/src/test/java/org/openmrs/module/patientflags/impl/FhirFlagServiceImplTest.java

```diff
@@ -0,0 +1,178 @@
+ */
+package org.openmrs.module.patientflags.impl;
+
+import ca.uhn.fhir.rest.api.server.IBundleProvider;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Flag;
+import org.hl7.fhir.r4.model.Reference;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.openmrs.Patient;
+import org.openmrs.module.fhir2.api.FhirGlobalPropertyService;
+import org.openmrs.module.fhir2.api.search.SearchQuery;
+import org.openmrs.module.fhir2.api.search.SearchQueryBundleProvider;
+import org.openmrs.module.fhir2.api.search.SearchQueryInclude;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.openmrs.module.patientflags.PatientFlag;
+import org.openmrs.module.patientflags.Priority;
+import org.openmrs.module.patientflags.Tag;
+import org.openmrs.module.patienttflags.dao.FhirFlagDao;
+import org.openmrs.module.patienttflags.impl.FhirFlagServiceImpl;
+import org.openmrs.module.patienttflags.search.param.FlagSearchParams;
+import org.openmrs.module.patienttflags.translators.PatientFlagTranslator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+/**
+ * The type Fhir flag service impl test.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class FhirFlagServiceImplTest {
+
+    private static final int START_INDEX = 0;
+
+    private static final int END_INDEX = 10;
+
+    String FLAG_UUID = "123xx34-623hh34-22hj89-23hjy5";
+
+    String FLAG_CODE = "FLAG MESSAGE";
+
+    String FLAG_CATEGORY = "FLAG CATEGORY";
+
+    String SUBJECT_DISPLAY = "Patient/ 123xx34-623hh34-22hj89-23hjy5 Patient Name 10000X";
+
+    Flag fhirFlag;
+
+    PatientFlag patientFlag;
+    
+    @Mock
+    private PatientFlagTranslator patientFlagTranslator;
+
+    @Mock
+    private FhirGlobalPropertyService globalPropertyService;
+
+
+    @Mock
+    private FhirFlagDao fhirFlagDao;
+    
+    @Mock
+    private SearchQueryInclude<Flag> searchQueryInclude;
+    
+    @Mock
+    private SearchQuery<PatientFlag, Flag, FhirFlagDao, PatientFlagTranslator, SearchQueryInclude<Flag>> searchQuery;
+    
+    private FhirFlagServiceImpl fhirFlagService;
+
+    /**
+     * Sets .
+     */
+    @Before
+    public void setup() {
+        fhirFlagService = new FhirFlagServiceImpl() {
+            /**
+             * @param object 
+             */
+            @Override
+            protected void validateObject(PatientFlag object) {
+            }
```

Comment:
Is this override strictly necessary? We shouldn't be invoking any of the endpoints that use it (create, update, patch).

Reply:
I removed this. I just added here as it is used in other services